x/vesting/keeper/msg_server.go

```diff
@@ -225,97 +224,22 @@ func (k Keeper) addGrant(
 }
 
 // transferClawback transfers unvested tokens in a ClawbackVestingAccount to
-// dest. Future vesting events are removed. Unstaked tokens are simply sent.
-// Unbonding and staked tokens are transferred with their staking state intact.
-// Account state is updated to reflect the removals.
+// dest address, updates the lockup schedule and removes future vesting events.
 func (k Keeper) transferClawback(
 	ctx sdk.Context,
 	va types.ClawbackVestingAccount,
 	dest sdk.AccAddress,
 ) error {
-	// Compute the clawback based on the account state only, and update account
+	// Compute clawback amount, unlock unvested tokens and remove future vesting events
 	updatedAcc, toClawBack := va.ComputeClawback(ctx.BlockTime().Unix())
 	if toClawBack.IsZero() {
 		return nil
 	}
-	addr := updatedAcc.GetAddress()
-	bondDenom := k.stakingKeeper.BondDenom(ctx)
-
-	// Compute the clawback based on bank balance and delegation, and update account
-	encumbered := updatedAcc.GetVestingCoins(ctx.BlockTime())
-	bondedAmt := k.GetDelegatorBonded(ctx, addr)
-	unbondingAmt := k.GetDelegatorUnbonding(ctx, addr)
-	bonded := sdk.NewCoins(sdk.NewCoin(bondDenom, bondedAmt))
-	unbonding := sdk.NewCoins(sdk.NewCoin(bondDenom, unbondingAmt))
-	unbonded := k.bankKeeper.GetAllBalances(ctx, addr)
-	updatedAcc, toClawBack = updatedAcc.UpdateDelegation(encumbered, toClawBack, bonded, unbonding, unbonded)
-
```

Comment:
To make sure I understand: you removed this logic because we do not allow delegation of unvested tokens. So, there is no reason to claw back any delegations.

Reply:
The PR description already answers this: ` remove unnecessary staking logic during clawback (as staking unvested tokens is not allowed ) `

Reply:
exactly!